blockly-core/core/ui/contract_editor/contract_editor.js

```diff
@@ -408,8 +408,15 @@ Blockly.ContractEditor.prototype.registerTestResetHandler = function (testResetH
   this.testResetHandler_ = testResetHandler;
 };
 
+/**
+ * Moves a given example block to the modal blockspace.
+ * @param {Blockly.Block} exampleBlock
+ */
 Blockly.ContractEditor.prototype.addExampleBlockFromMainBlockSpace = function(exampleBlock) {
```

Comment:
Are there any cases where this won't get called for an example block? I assume the answer is no. i.e. I assume this gets called both when adding a new example and when coming back to the level later and loading an existing example. This is why we can't get away with not persisting to xml, right?